### DISCUSSION TITLE
Welcome to UNES Discussions!

### DISCUSSION THREAD

**Discussion by ls9512** [ls9512/UNES#2] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by realjoecracker** [comment]
Have you attempted to implement mapper 30? It’s the default mapper for games made in  [NES Maker](https://www.thenew8bitheroes.com/) one game you can test is a popular demo known as [Troll Burner](https://theretroverse.com/product/troll-burner/). Also, have you gotten to where you can access options such as save states and rewind?

**Reply by ls9512** [reply]
Sorry for not being able to provide support for Mapper 30 for the time being, firstly because [Emulator.NES](https://github.com/Xyene/Emulator.NES), the upstream project of this project, does not provide this feature, and secondly because I could not find a valid reference implementation of Mapper 30 and specific format definition information.

The real-time state saving function has been implemented. If necessary, the multi-archive storage function can be easily extended. If the performance is not considered, the rewinding function based on a certain time interval can also be extended by using the real-time archive data. However, this project is only based on A Unity port of other projects, only to explore the possibility of running other platform games nested within Unity, there are no plans to develop other functions for the time being.

Of course if you or anyone else is willing to provide some usable improvements, PRs are welcome.
